src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
>do you agree that executing the "for (COUNT_T i = 0; i < length; i ++)>Attribute::SetBlittableCaValue" after it instead of before is irrelevant?

Yes, my recommended change is contingent on the ordering of those two operations being interchangable.

There is a strange interaction there in that SetBlittableCaValue() is modifying the bAllBlittableCA boolean which then gates the next "if". I need to take a closer look at that.